app/kubernetes/views/cluster/clusterController.js

```diff
@@ -6,17 +6,41 @@ import { KubernetesResourceReservation } from 'Kubernetes/models/resource-reserv
 
 class KubernetesClusterController {
   /* @ngInject */
-  constructor($async, Authentication, Notifications, KubernetesNodeService, KubernetesApplicationService) {
+  constructor($async, $state, Authentication, Notifications, LocalStorage, KubernetesNodeService, KubernetesApplicationService, KubernetesComponentStatusService) {
     this.$async = $async;
+    this.$state = $state;
     this.Authentication = Authentication;
     this.Notifications = Notifications;
+    this.LocalStorage = LocalStorage;
     this.KubernetesNodeService = KubernetesNodeService;
     this.KubernetesApplicationService = KubernetesApplicationService;
+    this.KubernetesComponentStatusService = KubernetesComponentStatusService;
 
     this.onInit = this.onInit.bind(this);
     this.getNodes = this.getNodes.bind(this);
     this.getNodesAsync = this.getNodesAsync.bind(this);
     this.getApplicationsAsync = this.getApplicationsAsync.bind(this);
+    this.getComponentStatus = this.getComponentStatus.bind(this);
+    this.getComponentStatusAsync = this.getComponentStatusAsync.bind(this);
+  }
+
+  async getComponentStatusAsync() {
+    try {
+      this.ComponentStatuses = await this.KubernetesComponentStatusService.get();
+
+      _.forEach(this.ComponentStatuses, (cs) => {
+        if (!cs.Healthy) {
+          this.hasUnhealthyComponentStatus = true;
+          return;
+        }
```

Comment:
Replace with `this.hasUnhealthyComponentStatus = _.find(this.ComponentStatuses, { Healthy: false }) ? true : false;`